tensorflow/python/framework/function.py

```diff
@@ -748,6 +748,9 @@ def foo(x, y):
   default graph and adds the definition of the function into the
   default graph. Because the addition of the function into the graph
   is deferred, the decorator can be used anywhere in the program.
+  
+  Definitions of function is frozen for the session after first session.run
```

Comment:
How about

Definitions of functions are frozen in a graph as soon as the graph is used to create a session.
Therefore, nodes using the function must be created in the graph before the corresponding session
is created.  

Maybe then add an example.